src/numerics/Func1.cpp

```diff
@@ -616,17 +658,19 @@ shared_ptr<Func1> newDiffFunction(shared_ptr<Func1> f1, shared_ptr<Func1> f2)
     if (isZero(f1)) {
         return newTimesConstFunction(f2, -1.);
     }
-
     if (f1->isIdentical(*f2)) {
         return make_shared<Const1>(0.);
     }
-    double c = f1->isProportional(*f2);
-    if (c != 0.) {
+    if (isConstant(f2)) {
+        return newPlusConstFunction(f1, -f2->c());
+    }
+    auto prop = isProportional(f1, f2);
+    if (prop.first) {
+        double c = prop.second;
```

Comment:
You can use a structured binding to make this a little bit nicer:
```suggestion
    auto [prop, c] = isProportional(f1, f2);
    if (prop) {
```